src/libmongoc/tests/test-mongoc-write-concern.c

```diff
@@ -595,10 +596,150 @@ test_write_concern_wtimeout_preserved (void)
    ASSERT (bson_iter_next (&child));
    ASSERT_CMPSTR (bson_iter_key (&child), "wtimeout");
    ASSERT_CMPINT64 (bson_iter_int64 (&child), ==, INT64_MAX);
-   
+
    mongoc_write_concern_destroy (write_concern);
 }
 
+/* callback that records write concern for commands */
+static void
+write_concern_count (const mongoc_apm_command_started_t *event)
+{
+   const bson_t *command = mongoc_apm_command_started_get_command (event);
+   bson_iter_t iter, iter_rec;
+
+   if (bson_iter_init (&iter, command)) {
+      int *sent_collection_w =
+         (int *) mongoc_apm_command_started_get_context (event);
+      if (bson_iter_find_descendant (&iter, "writeConcern.w", &iter_rec)) {
+         *sent_collection_w = bson_iter_int32 (&iter_rec);
+      } else {
+         *sent_collection_w = 1; /* no write concern sent. default used */
```

Comment:
I suggest using `MONGOC_WRITE_CONCERN_W_DEFAULT`. That constant is used in write concern objects to indicate not to append anything to the command (so it shouldn't appear in the command). This way, this won't accidentally match a sent acknowledged write concern.